contributors/design-proposals/node/troubleshoot-running-pods.md

```diff
@@ -106,13 +100,91 @@ subsequently be used to reattach and is reported by `kubectl describe`.
 
 ### Kubernetes API Changes
 
-#### Chosen Solution: "exec++"
+There has been much discussion about how this fits best into the Kubernetes API.
+The consensus is for an imperative "debug this pod" action whereby the kubelet
+creates a new, temporary container in a pod on command. SIG Node would like to
+avoid new dependencies in the kubelet, so this will be implemented in the Core
+API. Three possible implementations follow, and additional implementations that
+were evaluated and dismissed are at the end of this document.
+
+All of the proposed solutions implement the user-level concept of a _Debug
+Container_ using the API-level concept of an _Ephemeral Container_. The API
+doesn't prescribe how an Ephemeral Container is used. It could conceivably see
+use other than Debug Containers, but we don't currently have other use cases.
+
+#### Chosen Solution: Subresource to Update PodStatus
+
+An Ephemeral Container is not part of the pod specification as it's not part of
+the declared state of the pod, but we describe it using the same primitives as
+in `PodSpec`. An `EphemeralContainer` contains a Spec, a Status and a Target:
+
+```
+// EphemeralContainer describes a container to attach to a running pod for troubleshooting.
+type EphemeralContainer struct {
+        metav1.TypeMeta `json:",inline"`
```

Comment:
This makes me think it is a top-level resource - no other sub-structs have their own `kind` fields - why this one?  Can you make sure to explain it in comments, if it is intended?

Reply:
It is a subresource, like scale.

I don't think this can support anything but POST, is that the intention?

Reply:
Should we include metadata, and force the user to copy the UID of the pod they are targeting, to prevent accidentally getting a different pod than they intended if there is a delete/recreate?

Reply:
We definitely need preconditions to be possible here.  Requiring resourceVersion and uid has been sufficient for all other apis so far.